module/transfer.py

```diff
@@ -0,0 +1,292 @@
+                        content_type = "text/vtt" if typ == "subtitles" else "video/mp4"
+                        s3_client.upload_file(
+                            file_path,
+                            s3_bucket,
+                            item_path,
+                            ExtraArgs={"ContentType": content_type},
+                        )
+                        m["needsTransfer"] = False
+                        m["url"] = item_path
+                        update_media_vars = {"mentor": mentor, "question": question}
+                        if tag == "en":
+                            update_media_vars["vtt_media"] = m
+                        if tag == "web":
+                            update_media_vars["web_media"] = m
+                        if tag == "mobile":
+                            update_media_vars["mobile_media"] = m
+                        update_media(MediaUpdateRequest(**update_media_vars))
+                        answer_media_migrate_update = {
+                            "question": question,
+                            "status": "DONE",
+                        }
+                        import_task_update_gql(
+                            ImportTaskUpdateGQLRequest(
+                                mentor=mentor,
+                                answerMediaMigrateUpdate=answer_media_migrate_update,
+                            )
+                        )
+
+                    except Exception as x:
+                        media_url = m.get("url", "")
+                        logging.error(f"Failed to upload video {media_url} to s3 {x}")
+                        logging.exception(x)
+                        raise x
+                    finally:
+                        try:
+                            remove(file_path)
+                        except:
+                            pass  # lambdas tmp files are not important
+                else:
+                    answer_media_migrate_update = {
+                        "question": question,
+                        "status": "DONE",
+                    }
+                    import_task_update_gql(
+                        ImportTaskUpdateGQLRequest(
+                            mentor=mentor,
+                            answerMediaMigrateUpdate=answer_media_migrate_update,
+                        )
+                    )
+        except Exception as e:
+            logging.error(
+                f"Failed to process media for answer with question {question}"
+            )
+            answer_failure = e
+            logging.exception(e)
+            import_task_update_gql(
+                ImportTaskUpdateGQLRequest(
+                    mentor=mentor,
+                    answerMediaMigrateUpdate={
+                        "question": question,
+                        "status": "FAILED",
+                        "errorMessage": str(e),
+                    },
+                )
+            )
+        if answer_failure:
```

Comment:
Won't this cause the entire transfer to stop if a single answer fails? The previous behavior was to continue transferring the other answers even if one fails, which I think is desirable

Reply:
you're right, i wanted to have it outside the loop